sympy/polys/tests/test_rootoftools.py

```diff
@@ -175,6 +175,17 @@ def test_CRootOf_is_complex():
     assert rootof(x**3 + x + 3, 0).is_complex is True
 
 
+def test_CRootOf_Algebraic():
+    assert rootof(sqrt(2) * x ** 3 + sqrt(3) * x ** 2 + sqrt(3) * x + cbrt(3), 0) == \
+        CRootOf(64*x**36 - 576*x**34 - 1152*x**33 + 1584*x**32 + 8640*x**31 + 8640*x**30 \
+        - 17280*x**29 - 57780*x**28 - 47520*x**27 + 68364*x**26 + 217080*x**25 \
+        + 205389*x**24 - 69012*x**23 - 428166*x**22 - 535572*x**21 - 246645*x**20 \
+        + 227448*x**19 + 542232*x**18 + 548208*x**17 + 354699*x**16 + 153900*x**15 \
+        + 40014*x**14 - 10692*x**13 - 28917*x**12 - 22356*x**11 - 12150*x**10 \
+        - 9720*x**9 - 7290*x**8 - 2916*x**7 - 486*x**6 + 81, 0)
+    raises(NotImplementedError, lambda: rootof(sqrt(2) * x ** 3 + sqrt(3) * x + cbrt(3), 0))
```

Comment:
This shouldn't raise `NotImplementedError`. It looks like there's a bug in `primitive_element` that causes it to raise when there's a zero in the inputs.

Reply:
Yes, But I was unable to find a fix for that bug so raised a `NotImplementedError` for the polynomial.

Reply:
It looks like the trouble occurs in this loop

https://github.com/sympy/sympy/blob/c1e1f50cb68caa4976b63714fbe7cb84c3b4a342/sympy/polys/numberfields/subfield.py#L368-L380

when `p` is linear, so `L == QQ`. In particular, the call to `_switch_domain()` seems to be messed up, since there isn't really anything to `inject()`.

Maybe a fix could be as simple as starting the loop with something like this?

```python
    for ext in extension[1:]:
        p = minimal_polynomial(ext, x, polys=True)
        if p.degree() == 1:
            coeffs.append(0)
            continue
        L = QQ.algebraic_field((p, ext))
```

I haven't tried it against all unit tests, but it seems to allow the case at hand to pass...

Reply:
Yeah, seems like it and also it is passing all the unit tests as well.
But with this change along with changes with this PR, many tests of `rootoftools.py` are raising `TypeError: cannot unpack non-iterable NoneType object` .

Reply:
I would rather try something like this:

    for ext in extension[1:]:
        if ext.is_Rational:
            coeffs.append(0)    # rational ext is not included in the expression of a primitive element
            reps.append(K.convert(ext))    # but it is included in reps
            continue
        p = minimal_polynomial(ext, x, polys=True)
        L = QQ.algebraic_field((p, ext))

Reply:
This change is working and all the unit tests are passing, so should i include these changes along with this PR or in a separate PR along with relevant tests?

Reply:
Let's fix the bug in a separate PR.

Reply:
> I would rather try something like this:
> 
> ```
> for ext in extension[1:]:
>     if ext.is_Rational:
>         coeffs.append(0)    # rational ext is not included in the expression of a primitive element
>         reps.append(K.convert(ext))    # but it is included in reps
>         continue
>     p = minimal_polynomial(ext, x, polys=True)
>     L = QQ.algebraic_field((p, ext))
> ```

`ext` should be sympified i guess because I'm getting an error `AttributeError: 'int' object has no attribute 'is_Rational'`.